Haml related branch

Terunaga opened the pull request:
# WHAT

Transferred controller files, view files, assets files, Gemfile, and other necessary files that were already made by the staff.
# WHY

The workload would significantly be reduced by using the files that were already made.

Terunaga committed:
Installed 'haml-rails' and 'erb2haml' gems.

Terunaga committed:
Transferred controller files, view files, asset files and other files that were already made by the staff.

sinagaki58 commented on app/controllers/proto_controller.rb:
```diff
@@ -0,0 +1,6 @@
+class ProtoController < ApplicationController
+  def index
```
prototypes_controllerの方がいいですね！

sinagaki58 commented on app/controllers/top_controller.rb:
```diff
@@ -0,0 +1,4 @@
+class TopController < ApplicationController
+  def index
```
prototypes_controllerのindexと役割被ってるのでtopはいらないです
むしろpopularで並び替える用のranking_controllerみたいなのを作った方がいいですね

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -0,0 +1,31 @@
+      %span.icon-bar
+      %span.icon-bar
+    %a.navbar-brand(href="/")
+      = image_tag("logo.png", :alt => "PROTO TYPE - SHARE THE PROTOTYPE", :height => "36px")
```
今後

```
:hoge => "hoge"
```

みたいなのは、

```
hoge: "hoge"
```

に直しましょう！

sinagaki58 commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,12 @@
+  %body
+    = render "layouts/header"
+    = yield
+    = render "layouts/footer"
```
空行

sinagaki58 commented on app/views/top/index.html.haml:
```diff
@@ -0,0 +1,171 @@
+      %a{:href => "#"} 5
+    %li
+      %a{"aria-label" => "Next", :href => "#"}
+        %span{"aria-hidden" => "true"} »
```
top_controllerがいらないのでこのビューもいらないですね

sinagaki58 commented on config/routes.rb:
```diff
@@ -4,6 +4,15 @@
+  get  '/user'              =>    'user#index'
+  get  '/user/new'          =>    'user#new'
+  get  '/user/edit'         =>    'user#edit'
+  get  '/login'             =>    'login#index'
```
ルーティングは全てresourcesでいきましょう！

Terunaga committed:
Deleted the top controller and the related files by "rails d controller top".

Terunaga committed:
Created prototypes controller with "rails g controller prototypes".

Also, created index.html.haml file and new.html.haml file. The contents in each file were directly copied from those of the respective files under app/views/proto directory.

Terunaga committed:
Deleted the proto controller and the related view files by "rails d controller proto".

These files were replaced by prototypes controller and the related view files.

Terunaga committed:
Created ranking controller and the related files by "rails g controller ranking".

Also, created index.html.haml file, which will show the rankings of the prototypes.

Terunaga committed:
Added show, update and create actions in user controller, and created show.html.haml file.

Terunaga committed:
Added show action in prototypes controller (that will show the individual prototype).

Terunaga committed:
Added, and edited the routings by using resources method.

Terunaga committed:
Changed the style at which some attributes are written down.

eg. :hoge => "hoge"   to   hoge: "hoge".

There are still many places where this style needs to be fixed, but I will fix them as I make a progress.

sinagaki58 commented on app/assets/javascripts/prototypes.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
coffee系消しちゃいましょう

sinagaki58 commented on app/assets/javascripts/ranking.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
これも消しましょう

sinagaki58 commented on app/assets/stylesheets/prototypes.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the prototypes controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
これも消しましょう

sinagaki58 commented on app/assets/stylesheets/ranking.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the ranking controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
これも消しましょう

sinagaki58 commented on app/helpers/ranking_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module RankingHelper
+end
```
helper今のとこいらないので消しちゃいましょう

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -0,0 +1,30 @@
+      %ul.nav.navbar-nav.navbar-right
+        %li
+          .btn.btn-primary.navbar-btn
+            %a{href: "/login"}
```
aタグはlink_toに変えましょう

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -0,0 +1,30 @@
+      //login
+      %ul.nav.navbar-nav.navbar-right
+        %li.dropdown
+          %a.dropdown-toggle{"aria-expanded" => "false", "data-toggle" => "dropdown", href: "#", role: "button"}
```
ハッシュロケットは撲滅しましょう

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -0,0 +1,30 @@
+            %span.caret
+          %ul.dropdown-menu{role: "menu"}
+            %li
+              %a{href: "#"} Profile Edit
```
この辺もlink_toですね

sinagaki58 commented on app/views/layouts/_header.html.haml:
```diff
@@ -0,0 +1,30 @@
+        %li
+          .btn.btn-primary.navbar-btn
+            %a{href: "/proto/new"}
+              New Proto
```
空行

sinagaki58 commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,12 @@
+  %body
+    = render "layouts/header"
+    = yield
+    = render "layouts/footer"
```
空行

sinagaki58 commented on app/views/login/index.html.haml:
```diff
@@ -0,0 +1,17 @@
+    %p 
+      Not a member? 
+      %a{:href => "user/new"} Sign up now
+      
```
link_toとハッシュロケット直しましょう

sinagaki58 commented on config/routes.rb:
```diff
@@ -4,6 +4,13 @@
+  resources :ranking, only: :index
+  resources :user, only: [:index, :show, :new, :edit, :create, :update]
+
+  get  '/login'             =>    'login#index'
```
loginは使わないかと思います

sinagaki58 commented on test/controllers/prototypes_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
test系は消しましょう。

Terunaga committed:
Deleted unnecessary files at this stage (coffee related files and helper directory).

Terunaga committed:
Changed a tag to link_to method.

Terunaga committed:
Deleted login routing, along with login controller and its view file.

Terunaga committed:
Deleted test directory.

Terunaga committed:
Added a blank line at the end of each file.

Terunaga committed:
Recreated the index file of login controller, and changed the hash style.

Terunaga committed:
Modified hash style.

Terunaga commented:
# WHAT
- Modified the hash style.
- Modified <a> tag to link_to method.

# WHY
- To get rid of the hash rocket.
- Link_to method is more flexible than an a tag.

sinagaki58 commented on app/views/layouts/_footer.html.haml:
```diff
@@ -0,0 +1,3 @@
+.container
+  %footer
+    %p &copy; PROTO SPACE 2015.
```
空行

Terunaga committed:
Added a blank line.

Terunaga commented:
# WHAT

Added a blank line.

# WHY

It creates an unnecessary trouble without it.
